src/bosh-director/lib/bosh/director/jobs/update_deployment.rb

```diff
@@ -344,6 +290,165 @@ def event_context(next_releases, previous_releases, next_stemcells, previous_ste
         context['after'] = after_objects
         context
       end
+
+      def manifest_hash
+        return @manifest_hash if @manifest_hash
+
+        logger.info('Reading deployment manifest')
+        @manifest_hash = YAML.load(raw_manifest_text)
+        logger.debug("Manifest:\n#{raw_manifest_text}")
+        @manifest_hash
+      end
+
+      def cloud_config_models
+        return @cloud_config_models if @cloud_config_models
+
+        if ignore_cloud_config?(manifest_hash)
+          warning = "Ignoring cloud config. Manifest contains 'networks' section."
+          logger.debug(warning)
+          @event_log.warn_deprecated(warning)
+          @cloud_config_models = nil
+        else
+          @cloud_config_models = Bosh::Director::Models::Config.find_by_ids(@cloud_config_ids)
+          if cloud_config_models.empty?
+            logger.debug('No cloud config uploaded yet.')
+          else
+            logger.debug("Cloud config:\n#{Bosh::Director::CloudConfig::CloudConfigsConsolidator.new(cloud_config_models).raw_manifest}")
+          end
+        end
+
+        @cloud_config_models
+      end
+
+      def runtime_config_models
+        return @runtime_config_models if @runtime_config_models
+
+        @runtime_config_models = Bosh::Director::Models::Config.find_by_ids(@runtime_config_ids)
+        if runtime_config_models.empty?
+          logger.debug("No runtime config uploaded yet.")
+        else
+          logger.debug("Runtime configs:\n#{Bosh::Director::RuntimeConfig::RuntimeConfigsConsolidator.new(runtime_config_models).raw_manifest}")
+        end
+        @runtime_config_models
+      end
+
+      def deployment_name
+        manifest_hash['name']
+      end
```

Comment:
Some of these helper methods seem to clearly be helpful in understanding the true nature/type of an object, e.g. there is a more generic variable named `@manifest_text` but the function `raw_manifest_text` better conveys the intent of the object. Why was this one useful?